### PR TITLE
feat: template cloud integration [ref Codeinwp/templates-cloud#98]

### PR DESCRIPTION
**Pull request by abaicus**
Closes #.

### Screenshots 

----

### Test instructions

---- 

### Checklist before the final review

- [ ] Included E2E or unit tests for the changes in this PR.
- [ ] Visual elements are not affected by independent changes.
- [ ] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [ ] It loads additional script in frontend only if it is required.
- [ ] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [ ] In case of deprecation, old blocks are safely migrated.
- [ ] It is usable in Widgets and FSE.
- [ ] Copy/Paste is working if the attributes are modified.
- [ ] PR is following [the best practices]()

### CODE REVIEW

**Comment by ineagu** [comment]
@abaicus I had a chance to review this and here are some thoughts based on the updated implementation: 

1. Since we moved adding sources in the settings, I would also move the initially suggested message to clarify what this is about : https://vertis.d.pr/i/RVz5VW (using links and adapting the design).

2. My initial idea was that the "My Library" link would still be displayed even if no sources are added and would act as a upsell/feature discovery. Considering the new implementation I see 2 ideas: 

a) Display a link at the top like : https://vertis.d.pr/i/6Lt1sR, when clicked, it would open on the right some text explaining this + a link to the settings ( is this possible?

b) Since most of users won't use this, we can display that link at the bottom, after the categories.


3. I guess a concern you also had and decided not to show the link if there are no sources is how useful this would be for Otter users, we can open up quite easily a lot more possibilities and make this more useful if for example we suggest users a doc in which we would share for the start a free collection ( I can easily do that with some material design patterns that John created + install TC server on demosites).

Eventually later on we can also encourage users to share their collections if they want or we can create some more, free or paid.

To recap : 

1. We need to fix
2. I need to be clear if it possible and what makes sense for you a or b
3. I need your thoughts on potential technical challenges + on the idea itself. If we skip this, we can move with 2 b)

**Comment by abaicus** [comment]
@ineagu 

1. Sounds good. I agree! 👍🏻 What should we be linking to for TC & Otter Agency? Should we link straight to the pricing, or just the landing page?

2. You mean displaying some content instead of the patterns on the right part of the modal? That should be possible. I think we can go with **a**, as the pattern categories list is pretty big, and might grow larger in the future. Then again, I agree with you that the feature might not be used that much, so **b** might make more sense. I think either positioning is valid, but I incline to **a**, seeing that that's where we're going to display the categories after you connect. 

For the content on the right, do you have anything specific in mind, or can we go with something similar to what we'd have on **#1**, plus the link to settings?

3. I don't see much technical challenge regarding this. We can simply predefine a source the user can remove, but it means that they would lose context if we treat it as a simple source that can be removed, i.e. the user will not get the empty state of the sources panel, or the link we discussed on **#2**.

**Comment by abaicus** [comment]
Another question for **1**: Should this description be displayed regardless if the user has sources already, or only if there are no sources set yet? Also, for the links, would there be any desired UTM parameters that we should add? 

Here is the updated state with the description + links:

image

**Comment by ineagu** [comment]
1. For links we can use for both : ?utm_source=wpadmin&utm_medium=admin&utm_campaign=templatecloud&utm_content=otter-blocks
2. We can display the link at the top and for the content on the right here is the design & text : https://claude.site/artifacts/21a78a31-161f-4309-8385-f54e147311ef

3. For the settings, we can use this link and text, before the existing notice : Check-out our curated list of sources that you can import [here](https://docs.themeisle.com/article/2191-templates-cloud-collections).

File: inc/class-main.php [hunk 1]
```diff
@@ -83,6 +83,8 @@ public function autoload_classes() {
 			'\ThemeIsle\GutenbergBlocks\Integration\Form_Email',
 			'\ThemeIsle\GutenbergBlocks\Server\Form_Server',
 			'\ThemeIsle\GutenbergBlocks\Server\Prompt_Server',
+			'\ThemeIsle\GutenbergBlocks\Template_Cloud',
```
Comment (HardeepAsrani) [inline inc/class-main.php:86]: We usually put namespace new features like this as part of [Plugins](https://github.com/Codeinwp/otter-blocks/tree/master/inc/plugins).

File: src/dashboard/components/template-cloud/Sources.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+				__('Source deleted', 'otter-blocks'),
+				{
+					isDismissible: true,
+					type: 'snackbar'
+				}
+			);
+		}).catch((e) => {
+			setError(e?.message ?? __('An unknown error occurred.', 'otter-blocks'));
+		});
+	};
```
Comment (HardeepAsrani) [inline src/dashboard/components/template-cloud/Sources.js:37]: If you've tested it and it's working then it's fine but in past apiFetch didn't support using DELETE method like this, and I had to do it like this instead:

```
		const response = await apiFetch({
			path: `${ window.hyve.api }/faq/${ id }`,
			method: 'POST',
			headers: {
				'X-HTTP-Method-Override': 'DELETE'
			}
		});
```

In my case, usually the DELETE and POST methods were calling the same API endpoint with different results. I suppose you're using a route that ONLY supports DELETE method and not POST method, it might be working fine.

**Review by HardeepAsrani — APPROVED** [review]
Looks good, left few observations.

**Comment by abaicus** [comment]
@HardeepAsrani thanks for the suggestions! I've addressed them 👍🏻

**Comment by ineagu** [comment]
@abaicus 

Let's also add the external link icon to  Premium designs and move on 3rd position
Use f18f icon for My Library

This to make it clear that is a feature and not an external link or that is not "locked"
